src/components/hierarchy-bar/hierarchy-bar-node/hierarchy-bar-node.component.html

```diff
@@ -28,14 +38,55 @@
             aria-label="Show children"
             role="button"
             class="hierarchy-bar-node-arrow"
+            [class.hierarchy-bar-node-arrow-readonly]="mode === 'readOnly'"
             [uxPopover]="popoverTemplate"
             [popoverContext]="{ node: node, popover: popover }"
             placement="bottom"
             popoverClass="hierarchy-bar-popover"
-            [showTriggers]="hierarchyBar.popoverShowTriggers"
-            [hideTriggers]="hierarchyBar.popoverHideTriggers"
+            [showTriggers]="mode !== 'readOnly' ? hierarchyBar.popoverShowTriggers : null"
```

Comment:
I think this should probably be `[]` instead of `null` on both the show and hide triggers. It is expecting an array so if we call any array functions anywhere this could throw any error if the value is `null`